keras_cv/bounding_box/converters.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Converter functions for working with bounding boxes.
+
+Usually bounding boxes is a 2D Tensor with shape [batch, 4]. The second dimension
+will contain 4 numbers based on 2 different formats.  In KerasCV, we will use the
+`corners` format, which is [LEFT, TOP, RIGHT, BOTTOM].
+
+In this file, provide utility functions for manipulating bounding boxes and converting
+their formats.
+"""
+
+import tensorflow as tf
+
+
+# Internal exception to propagate the fact images was not passed to a converter that
+# needs it
+class RequiresImagesException(Exception):
+    pass
```

Comment:
This will never be surfaced to users, right?

In general we should never raise a custom exception to users. But if we're sure that it will be caught internally and reraised as a regular exception this is fine.

Reply:
This will never be raised to users, it is an implementation detail only.